src/taskgraph/transforms/task.py

```diff
@@ -142,6 +142,15 @@ def _run_task_suffix():
         Optional("run-on-projects"): optionally_keyed_by("build-platform", [str]),
         Optional("run-on-tasks-for"): [str],
         Optional("run-on-git-branches"): [str],
+        # The `shipping_phase` attribute, defaulting to None. This specifies the
+        # release promotion phase that this task belongs to.
+        Optional("shipping-phase"): Any(
```

Comment:
This is the one change I made compared to gecko. We don't want to fail if shipping-phase is missing so it's not `Required` anymore. 

https://searchfox.org/mozilla-central/rev/404408660a4d976e2ac25881cb1e1f2712f2d430/taskcluster/gecko_taskgraph/transforms/task.py#169